subsys/dfu/boot/mcuboot.c

```diff
@@ -34,6 +34,14 @@
 #define BOOT_HEADER_MAGIC_V1 0x96f3b83d
 #define BOOT_HEADER_SIZE_V1 32
 
+#ifdef CONFIG_FLASH_LOAD_OFFSET
+#define FIXED_PARTITION_IS_RUNNING_APP_PARTITION(label)	\
+	 (FIXED_PARTITION_OFFSET(label) == CONFIG_FLASH_LOAD_OFFSET)
+#else
+#define FIXED_PARTITION_IS_RUNNING_APP_PARTITION(label)	\
+	 (FIXED_PARTITION_OFFSET(label) == 0)
+#endif
+
```

Comment:
Not convinced with this, should instead use `DT_CHOSEN()` to check which it is.

Reply:
I based this off what is in `zephyr/subsys/mgmt/mcumgr/grp/img_mgmt` - this is the method used there to get the running app partition. I could instead use `DT_FIXED_PARTITION_ID(DT_CHOSEN(zephyr_code_partition))` to get the flash area ID - that would remove a lot of the code added in this PR.

I think a fallback to a default partition will still be needed for the tests to pass.

Reply:
PR updated to use `DT_CHOSEN()`, the additions are significantly reduced now. Please check when you can @nordicjm @de-nordic